examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2097,200 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
+					"day", bson.D{{"$dayOfWeek", "$date"}},
+				}}},
+				{"count", bson.D{{"$sum", "$items.quantity"}}},
+			}}},
+			bson.D{{"$project", bson.D{
+				{"dayOfWeek", "$_id.day"},
+				{"numberSold", "$count"},
+				{"_id", 0},
+			}}},
+			bson.D{{"$sort", bson.D{{"numberSold", 1}}}},
+		}
+
+		cs, err := coll.Aggregate(ctx, pipeline)
+		require.NoError(t, err)
+		defer cs.Close(ctx)
+
+		ok := cs.Next(ctx)
+		next := cs.Current
+
+		// End Aggregation Example 2
+
+		require.True(t, ok)
+		require.NoError(t, err)
+		require.NotEqual(t, len(next), 0)
+	}
+	{
+		// Start Aggregation Example 3
+		pipeline := bson.A{
+			bson.D{{"$unwind", "$items"}},
+			bson.D{{"$group", bson.D{
+				{"_id", bson.D{
+					{"day", bson.D{{"$dayOfWeek", "$date"}}},
+				}},
+				{"items_sold", bson.D{{"$sum", "$items.quantity"}}},
+				{"revenue", bson.D{{"$sum", bson.D{
+					{"$multiply", bson.A{"$items.quantity", "$items.price"}},
+				}}}},
+			}}},
+			bson.D{{"$project", bson.D{
+				{"day", "$_id.day"},
+				{"revenue", 1},
+				{"items_sold", 1},
+				{"discount", bson.D{{"$cond", bson.D{{"if", bson.D{
+					{"$lte", bson.A{"$revenue", 250}},
+					{"then", 25},
+					{"else", 0},
+				}}}}}},
+			}}},
+		}
+
+		cs, err := coll.Aggregate(ctx, pipeline)
+		require.NoError(t, err)
+		defer cs.Close(ctx)
+
+		ok := cs.Next(ctx)
+		next := cs.Current
+
+		// End Aggregation Example 3
+
+		require.True(t, ok)
+		require.NoError(t, err)
+		require.NotEqual(t, len(next), 0)
+	}
+	{
+		// Start Aggregation Example 4
+		pipeline := bson.A{
+			bson.D{{"$lookup", bson.D{
+				{"from", "air_airlines"},
+				{"let", bson.D{{"constituents", "$airlines"}}},
+				{"pipeline", bson.A{"$match", bson.D{
+					{"$expr", bson.D{{"$in", bson.A{"$name", "$$constituents"}}}},
+				}}},
+				{"as", "airlines"},
+			}}},
+			bson.D{{"$project", bson.D{
+				{"_id", 0},
+				{"name", 1},
+				{"airlines", bson.D{{"$filter", bson.D{
+					{"input", "$airlines"},
+					{"as", "airline"},
+					{"cond", bson.D{{"$eq", bson.A{"$$airline.country", "Canada"}}}},
+				}}}},
+			}}},
+		}
+
+		cs, err := coll1.Aggregate(ctx, pipeline)
+		require.NoError(t, err)
+		defer cs.Close(ctx)
+
+		ok := cs.Next(ctx)
+		next := cs.Current
+
+		// End Aggregation Example 4
+
+		require.True(t, ok)
+		require.NoError(t, err)
+		require.NotEqual(t, len(next), 0)
+	}
+}
+
+// RunCommandExamples contains examples of RunCommand operations.
+func RunCommandExamples(t *testing.T, db *mongo.Database) {
+	ctx := context.Background()
+
+	{
+		// Start RunCommand Example 1
+		res := db.RunCommand(ctx, bson.D{{"buildInfo", 1}})
+		// End RunCommand Example 1
+
+		err := res.Err()
+		require.NoError(t, err)
+	}
+	{
+		// Start RunCommand Example 2
+		res := db.RunCommand(ctx, bson.D{{"collStats", "restaurants"}})
+		// End RunCommand Example 2
+
+		err := res.Err()
+		require.NoError(t, err)
+	}
+}
+
+// IndexExamples contains examples of Index operations.
+func IndexExamples(t *testing.T, db *mongo.Database) {
```

Comment:
For readability, I'd recommend splitting out the index creation into multiple lines. An example of how the second invocation could be refactored: https://play.golang.org/p/jYsAxwzvMwo.

Reply:
Great! Thanks for the example. Did something very similar for the first invocation, too.